File: spring-boot-rest-api/src/main/java/br/com/uabrestingaseca/biblioteca/model/Pendencia.java
package br.com.uabrestingaseca.biblioteca.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "pendencia")
public class Pendencia {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @NotNull(message = "Usuário deve ser informado")
    private Usuario usuario;

    @NotNull(message = "Valor deve ser informado")
    private BigDecimal valor;

    private String descricao;

    @ManyToOne
    private Emprestimo emprestimo;

    @Column(name = "data_hora_lancamento")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dataHoraLancamento;

    @Column(name = "data_hora_pagamento")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dataHoraPagamento;

}

File: biblioteca-react/src/model/Pendencia.ts
import { Emprestimo, Usuario } from ".";
import DateTimeUtil from "../util/DateTimeUtil";

export class Pendencia{

    constructor(
        public id?:number,
        public usuario?: Usuario,
        public valor?: number,
        public descricao?: string,
        public emprestimo?: Emprestimo,
        public dataHoraLancamento?: Date,
        public dataHoraPagamento?: Date
    ){}

    public static createFromData(data : any):Pendencia{
        const pendencia = new Pendencia();
        pendencia.id = data.id;
        pendencia.usuario = Usuario.createFromData(data.usuario);
        pendencia.valor = Number(data.valor).valueOf();
        pendencia.descricao = data.descricao;
        pendencia.emprestimo = Emprestimo.createFromData(data.emprestimo);
        pendencia.dataHoraLancamento = DateTimeUtil.fromAPIDateTime(data.dataHoraLancamento);
        pendencia.dataHoraPagamento = DateTimeUtil.fromAPIDateTime(data.dataHoraPagamento);
        return pendencia;
    }

    public foiPaga():boolean{
        return this.dataHoraPagamento? true : false;
    }
    
}